### PR TITLE
[FEATURE][#52]Remove the heap allocation from non-single queues

### PR DESCRIPTION
**Pull request by mamaicode**
#52 
Draft

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -57,7 +57,7 @@ extern crate alloc;
 #[cfg(feature = "std")]
 extern crate std;
 
-use alloc::boxed::Box;
+// use alloc::boxed::Box;
```
Comment (notgull) [inline src/lib.rs:60]: Remove this comment

File: src/lib.rs [hunk 2]
```diff
@@ -103,8 +103,8 @@ impl<T> RefUnwindSafe for ConcurrentQueue<T> {}
 
 enum Inner<T> {
```
Comment (notgull) [inline src/lib.rs:104]: Add `#[allow(clippy::large-enum-variant)]` to this enum to silence clippy.

**Review by notgull — CHANGES_REQUESTED** [review]
Overall looks good to me, but I'd like a second opinion.

@smol-rs/admins Thoughts?

**Review by fogti — APPROVED** [review]

**Comment by fogti** [comment]
NOTE: use squash merging / better commit message.

**Review by notgull — APPROVED** [review]

**Pull request merged**